### ISSUE TITLE
Could I skip typing password each time of starting a block?

### ISSUE THREAD

**Issue by 1c383r9** [SelfControlApp/selfcontrol#500]
1. Is there a helper application or some permission escalation tools to avoid the password inputting procedure ?
2. There's nothing to be displayed in the preference panel of the app. Both section has nothing to show, only an empty space there. Is this a bug?

Thanks.

**Comment by sankara** [comment]
@cstigler  - I want to explore this. I use SelfControl app in my company provided Mac. And I can request admin privileges only for x amount of days. Do you have any suggestions on any implementation approaches you have looked at already?

**Comment by cstigler** [comment]
Hi @sankara! I haven't looked into this one. Overall, I'm mostly fine with it requiring a password to start - it reinforces that this is a serious/irreversible action and prevents somebody from starting a block on your laptop if you  you leave it unlocked.

That said, obviously in a case like yours where you have only limited admin access, I can see why it's annoying. The most likely approaches I can think of are:

1) set SUID on the helper tool (which needs admin to install itself) so there's no admin access required after that

2) install the helper tool once, make it an always-on daemon or similar, never uninstall it so there's no need to reinstall it (and use admin)

that said I haven't done any real research so not sure if there's a nicer way!

**Comment by sankara** [comment]
@cstigler Appreciate the informative reply. I ended up getting permanent admin access.

I do understand your rationale for asking for admin password to emphasize that it's an irreversible action. I'm watching this thread too. If there's enough people asking for it, I'll look into implementing a solution. 

Thanks for this great utility btw. Sometimes we all need some extra help in focusing!

**Comment by Pandoks** [comment]
I created a soft solution to this if you wanted to use the functionality to schedule blocking sessions: [Hammer Control](https://github.com/Pandoks/hammer-control). It automatically inputs your password and starts the session. There's still a quick flicker of the prompt though, but it's the best that I could do.

**Comment by dnecklesportfolio** [comment]
>  Overall, I'm mostly fine with it requiring a password to start - it reinforces that this is a serious/irreversible action and prevents somebody from starting a block on your laptop if you you leave it unlocked.

Yes while I understand that keeping the seriousness of the action in mind is good. Maybe there could be an option for it, as ridiculous as that sounds for a small issue.
Why? I know it sounds like a minor detail but I don't want to be delayed having to input my password. Assuming that the app provides me a focused state, I want to get there as soon as possible. I rediscovered how useful the app could be.

In any case it's not a deal breaker, I will deal with it.
